client/asset/ltc/spv.go

```diff
@@ -0,0 +1,1049 @@
+						Height: lastBlock.Height,
+					}:
+					default:
+					}
+				}
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+	return ch
+}
+
+// secretSource is used to locate keys and redemption scripts while signing a
+// transaction. secretSource satisfies the txauthor.SecretsSource interface.
+type secretSource struct {
+	w           *wallet.Wallet
+	chainParams *ltcchaincfg.Params
+}
+
+// ChainParams returns the chain parameters.
+func (s *secretSource) ChainParams() *ltcchaincfg.Params {
+	return s.chainParams
+}
+
+// GetKey fetches a private key for the specified address.
+func (s *secretSource) GetKey(addr ltcutil.Address) (*btcec.PrivateKey, bool, error) {
+	ma, err := s.w.AddressInfo(addr)
+	if err != nil {
+		return nil, false, err
+	}
+
+	mpka, ok := ma.(ltcwaddrmgr.ManagedPubKeyAddress)
+	if !ok {
+		e := fmt.Errorf("managed address type for %v is `%T` but "+
+			"want waddrmgr.ManagedPubKeyAddress", addr, ma)
+		return nil, false, e
+	}
+
+	privKey, err := mpka.PrivKey()
+	if err != nil {
+		return nil, false, err
+	}
+
+	k, _ /* pub */ := btcec.PrivKeyFromBytes(privKey.Serialize())
+
+	return k, ma.Compressed(), nil
+}
+
+// GetScript fetches the redemption script for the specified p2sh/p2wsh address.
+func (s *secretSource) GetScript(addr ltcutil.Address) ([]byte, error) {
+	ma, err := s.w.AddressInfo(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	msa, ok := ma.(ltcwaddrmgr.ManagedScriptAddress)
+	if !ok {
+		e := fmt.Errorf("managed address type for %v is `%T` but "+
+			"want waddrmgr.ManagedScriptAddress", addr, ma)
+		return nil, e
+	}
+	return msa.Script()
+}
+
+// spvService embeds ltcsuite neutrino.ChainService and translates types.
+type spvService struct {
+	*neutrino.ChainService
+}
+
+var _ btc.SPVService = (*spvService)(nil)
+
+func (s *spvService) GetBlockHash(height int64) (*chainhash.Hash, error) {
+	ltcHash, err := s.ChainService.GetBlockHash(height)
+	if err != nil {
+		return nil, err
+	}
+	return convertHashToBTC(*ltcHash), nil
+}
+
+func (s *spvService) BestBlock() (*headerfs.BlockStamp, error) {
+	bs, err := s.ChainService.BestBlock()
+	if err != nil {
+		return nil, err
+	}
+	return &headerfs.BlockStamp{
+		Height:    bs.Height,
+		Hash:      *convertHashToBTC(bs.Hash),
+		Timestamp: bs.Timestamp,
+	}, nil
+}
+
+func (s *spvService) Peers() []btc.SPVPeer {
+	rawPeers := s.ChainService.Peers()
+	peers := make([]btc.SPVPeer, len(rawPeers))
+	for i, p := range rawPeers {
+		peers[i] = p
+	}
+	return peers
+}
+
+func (s *spvService) GetBlockHeight(h *chainhash.Hash) (int32, error) {
+	return s.ChainService.GetBlockHeight(convertHashToLTC(*h))
+}
+
+func (s *spvService) GetBlockHeader(h *chainhash.Hash) (*wire.BlockHeader, error) {
+	hdr, err := s.ChainService.GetBlockHeader(convertHashToLTC(*h))
+	if err != nil {
+		return nil, err
+	}
+	return &wire.BlockHeader{
+		Version:    hdr.Version,
+		PrevBlock:  *convertHashToBTC(hdr.PrevBlock),
+		MerkleRoot: *convertHashToBTC(hdr.MerkleRoot),
```

Comment:
Valid explicit conversions since one is representable by the other (same underlying type):

```go
		PrevBlock:  chainhash.Hash(hdr.PrevBlock),
		MerkleRoot: chainhash.Hash(hdr.MerkleRoot),
```